### PR TITLE
Barebones VR support, take two

### PR DESCRIPTION
**Pull request by kyle-emmerich**
Fixed the issue with a line being commented out! This is barebones VR support, just the bare minimum to somewhat prevent nausea when using VR on ROBLOX. EDIT: Follow is good, but avoid Watch.

### CODE REVIEW

**Comment by kyle-emmerich** [comment]
I tested and made sure that the camera behaves as it did before changing internal state handling. Seems to work okay, should we move ahead?

**Comment by kyle-emmerich** [comment]
Fixed up Follow camera type.

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx [hunk 1]
```diff
@@ -138,7 +139,7 @@ local function CreateFollowCamera()
 						local lastVec = -(self.LastCameraTransform.p - subjectPosition)
 						local y = findAngleBetweenXZVectors(lastVec, self:GetCameraLook())
 						-- Check for NaNs
-						if IsFinite(y) and math.abs(y) > 0.0001 then
+						if IsFinite(y) and math.abs(y) > 0.0001 and not UserInputService.IsVREnabled then
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: check should either be in the elseif statement (no reason to do all the math to just to bail)

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx [hunk 2]
```diff
@@ -72,6 +73,8 @@ end
 local function CreateCamera()
 	local this = {}
 
+	this.cframe = workspace.CurrentCamera and workspace.CurrentCamera.CoordinateFrame or CFrame.new()
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx]: what do we gain by abstracting cframe? It seems like it just adds a layer of direction that could be broken, and I'm not sure if there are any tangible performance benefits.

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx (hunk 1 above)
Comment (kyle-emmerich) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: True enough, will fix

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx (hunk 2 above)
Comment (kyle-emmerich) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx]: The reasoning is that previously, the camera scripts would just work off the existing state of the camera object. If your VR offset is modifying that, your state is now invalid unless you roll it back somehow, which is prone to floating point precision errors and isn't very friendly if you need to mix the transforms in a different way. It's definitely necessary if we want reasonable VR code.

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx [hunk 3]
```diff
@@ -134,7 +135,7 @@ local function CreateFollowCamera()
 								self.RotateInput = self.RotateInput + Vector2.new(y * percent, 0)
 							end
 						end
-					elseif not (isInFirstPerson or userRecentlyPannedCamera) then
+					elseif not (isInFirstPerson or userRecentlyPannedCamera) and UserInputService.IsVREnabled then
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: think this is missing a not here

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript.rbxmx [hunk 4]
```diff
@@ -138,6 +138,8 @@ end
 local function Update()
 	if EnabledCamera then
 		EnabledCamera:Update()
+		EnabledCamera:ApplyCameraCFrame()
+		EnabledCamera:ApplyVROffset()
```
Comment (BinaryResolved) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript.rbxmx]: Is :ApplyVROffset() a Lua function or a ROBLOX made one?

Comment (kyle-emmerich) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript.rbxmx]: It's a Lua function, you can find it in RootCamera. It basically just applies the UserHeadCFrame on top of the current camera CFrame, but the plan was for individual cameras to modify it for more special handling.

I would love to see these camera modules become more object-oriented. Right now most of them just repeat the same code.

Comment (BinaryResolved) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript.rbxmx]: I am thinking about getting a VR headset. What kind are you using?

Comment (kyle-emmerich) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript.rbxmx]: The Oculus Rift DK2 works best so far. The CV1 probably works, but of course we can't really verify yet.

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx (hunk 3 above)
Comment (kyle-emmerich) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: Good catch, I'll stop with the derp commits some time soon

**Pull request merged**